source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -78,11 +75,10 @@ SINGLETON_MANAGER_REGISTRATION(date_provider);
 SINGLETON_MANAGER_REGISTRATION(route_config_provider_manager);
 SINGLETON_MANAGER_REGISTRATION(scoped_routes_config_provider_manager);
 
-Network::FilterFactoryCb
-HttpConnectionManagerFilterConfigFactory::createFilterFactoryFromProtoTyped(
-    const envoy::extensions::filters::network::http_connection_manager::v3alpha::
-        HttpConnectionManager& proto_config,
-    Server::Configuration::FactoryContext& context) {
+std::tuple<std::shared_ptr<Http::TlsCachingDateProviderImpl>,
+           std::shared_ptr<Router::RouteConfigProviderManager>,
+           std::shared_ptr<Router::ScopedRoutesConfigProviderManager>>
```

Comment:
nit: I would just return a struct with 3 members. It is easier to read and will be easy to capture in the lambda, etc.